gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -145,22 +145,6 @@ impl ClnRpcService {
                     handle.await?
                 },
             )
-            // Shutdown the plugin when lightningd is shutting down or when the plugin is stopped
-            // via `plugin stop` command. There's a chance that the subscription is never called in
-            // case lightningd crashes or aborts.
-            // For details, see documentation for `shutdown` event notification:
-            // https://lightning.readthedocs.io/PLUGINS.html?highlight=shutdown#shutdown
-            .subscribe(
```

Comment:
Simple. I like it!

How come we no longer need to subscribe to `shutdown` event?

Reply:
If we receive a `shutdown` then  lightningd will kill our plugin regardless of whether we handle if AFAIK.

Our problem is when lightningd is killed and doesn't seem to shutdown our plugin at all. But this is solved by awaiting `plugin.join()` and shutting down when that resolves.